### ISSUE TITLE
Transient directives menu does not show user-defined directives

### ISSUE THREAD

**Issue by winston-churchil** [karthink/gptel#42]
If you add a directive to `gptel-directives` with M-x customize-option or with `(customize-set-option ...)` the directive is not available to be set in the `h Set directives for chat` submenu of `gptel-menu`. Unless I'm being stupid.

**Comment by karthink** [comment]
Indeed, this is the current behavior because of limitations with the last tagged release of transient. 

See https://github.com/karthink/gptel/blob/master/gptel-transient.el#L171.

The commented out code generates the directives dynamically from `gptel-directives`, and it actually works fine (once uncommented) if you use the latest git version of Transient. But it breaks completely for people using tagged releases of Transient (such as from MELPA stable or ELPA or the built-in Emacs version), so I've had to disable it until there's a new release.

See also the documentation for `gptel-directives`, which explains this issue.

**Comment by olgeni** [comment]
Is there a way to set the task too? Like code-generation etc? 👀

**Comment by karthink** [comment]
> Is there a way to set the task too? Like code-generation etc? 👀

You would use the system prompt for this, bound to h in the transient menu (C-u M-x gptel-send or M-x gptel-menu).

**Comment by psionic-k** [comment]
Newest tagged transient is out.  Let's bump the transient version requirement, tag a new release of gptel, and let this close.

**Comment by karthink** [comment]
v0.4 looks good, will update gptel soon.

**Comment by karthink** [comment]
I've pushed a fix in 706ad703. Please test. To make sure the dynamic menus are
generated correclty, it will help further to test with a list of directives with
partially overlapping names, such as

``` emacs-lisp
(setq gptel-directives
      '((programming . "You are a large language model and a careful programmer. Provide code and only code as output without any additional text, prompt or note.")
        (prose . "You are a large language model and a writing assistant. Respond concisely.")
        (default-chat . "You are a large language model and a conversation partner. Respond concisely.")
        (default . "You are a large language model living in Emacs and a helpful assistant. Respond concisely.")))
```

**Comment by karthink** [comment]
I'm assuming this works as intended with the new transient version. Please re-open if this is not the case.
